react/components/Conditions/index.tsx

```diff
@@ -0,0 +1,265 @@
+                        isFullWidth={isFullWidth}
+                        onChangeStatement={newStatement => {
+                          const newStatements = statements.map(
+                            (statement, idx) =>
+                              idx === statementIndex
+                                ? newStatement
+                                : statement
+                          )
+                          onChangeStatements(newStatements)
+                        }}
+                        options={uniqueBasedOptions}
+                        subjectPlaceholder={subjectPlaceholder}
+                        statement={statement}
+                        labels={labels}
+                      />
+                    </div>
+
+                    {canDelete &&
+                      (!isFullWidth ? (
+                        <div
+                          className="ma3 c-muted-2 pointer hover-c-danger"
+                          onClick={() =>
+                            handleRemoveStatement(statementIndex)
+                          }>
+                          <IconClose size={25} />
+                        </div>
+                      ) : (
+                        <div className="tr mh3">
+                          <Button
+                            variation="tertiary"
+                            collapseRight
+                            size="small"
+                            onClick={handleRemoveStatement}>
+                            <div className="dib">
+                              <IconClose className="c-on-action-primary" />
+                            </div>
+
+                            <div
+                              className="dib mb1 v-mid"
+                              style={{ lineHeight: '10px' }}>
+                              {labels.delete}
+                            </div>
+                          </Button>
+                        </div>
+                      ))}
+                  </div>
+
+                  {statementIndex !== statements.length - 1 && (
+                    <Separator
+                      label={
+                        operator === 'all'
+                          ? labels.operatorAnd || 'and'
+                          : labels.operatorOr || 'or'
+                      }
+                    />
+                  )}
+                </div>
+              )
+            })}
+          </div>
+        )}
+        <Separator />
+        <div className="mv5 mh3">
+          <ButtonWithIcon
+            variation="tertiary"
+            collapseLeft
+            size="small"
+            icon={<IconPlus solid size={MEDIUM_ICON_SIZE} />}
+            disabled={!canAddNewCondition()}
+            onClick={handleAddNewCondition}>
+            {labels.addNewCondition}
+          </ButtonWithIcon>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+// this is so styleguidist can pick the prop types reference
+Conditions.propTypes = propTypes
```

Comment:
is this really necessary in all .tsx index components? I did this so the props and their types can appear in styleguidist example page

Reply:
Not really, you can just define the `type`. Looks cleaner

Reply:
the type is defined in the begining of the file. But just the type isn't enough for styleguidist page :/ when it builds the page it checks for `${Component}.propTypes`, and when It's a React.FC with just types it does not find that and the example page breaks.

Reply:
Ow, I got it wrong. I was thinking that it was an internal component, that its docs are not shown on `styleguide` (sorry). In that case, it's not necessary 😅